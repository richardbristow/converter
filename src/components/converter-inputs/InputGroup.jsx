import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import FilterDropdown from './FilterDropdown';

const StyledInputGroup = styled.div`
  grid-area: ${props => props.name};
  display: grid;
  grid-template-columns: 1fr;
  grid-auto-rows: 50px;
`;

const InputGroup = ({
  name, textValue, dropdownValue, options, handleChange, conversionType,
}) => (
  <StyledInputGroup name={name}>
    <input name={`${name}Input`} value={textValue} type="text" onChange={e => handleChange(conversionType, e)} />
    <select name={`${name}Unit`} value={dropdownValue} onChange={e => handleChange(conversionType, e)}>
      {options.map(({ mathName, displayName }) =>
        <option key={`${name}-${mathName}`} value={mathName}>{displayName}</option>)}
    </select>
    <FilterDropdown
      options={options}
      name={name}
      handleChange={handleChange}
      conversionType={conversionType}
      dropdownValue={textValue}
    />
  </StyledInputGroup>
);

InputGroup.propTypes = {
  name: PropTypes.string.isRequired,
  textValue: PropTypes.string.isRequired,
  dropdownValue: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({
    displayName: PropTypes.string.isRequired,
    mathName: PropTypes.string.isRequired,
  })).isRequired,
  handleChange: PropTypes.func.isRequired,
  conversionType: PropTypes.string.isRequired,
};

export default InputGroup;
