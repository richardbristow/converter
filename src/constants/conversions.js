import length from './units/length';
import surfaceArea from './units/surfaceArea';

const conversions = {
  length,
  surfaceArea,
  // volume,
  // liquidVolume,
  // angles,
  // time,
  // frequency,
  // mass,
  // electricCurrent,
  // temperature,
  // amountOfSubstance,
  // luminousIntensity,
  // currency,
  // force,
  // energy,
  // power,
  // pressure,
  // electricityAndMagnetism,
  // binary,
};

export default conversions;
