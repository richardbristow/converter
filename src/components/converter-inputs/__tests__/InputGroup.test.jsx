import React from 'react';
import { shallow } from 'enzyme';
import InputGroup from '../InputGroup';
import conversions from '../../../constants/conversions';

const handleChange = jest.fn();

const props = {
  convert: {
    handleChange,
    inputLeft: '',
    inputRight: '',
  },
  conversions,
  conversionType: 'surfaceArea',
};

describe('InputGroup', () => {
  it('should render 2 dropdown menus', () => {
    const wrapper = shallow(<InputGroup {...props} />);
    expect(wrapper.find('Dropdown')).toHaveLength(2);
  });

  it('should mount dropdowns with default values', () => {
    const {
      initialUnitLeft: defaultUnitLeft,
      initialUnitRight: defaultUnitRight,
    } = conversions.surfaceArea;
    const wrapper = shallow(<InputGroup {...props} />);
    expect(wrapper.find('Dropdown.unitLeft').props().value).toBe(defaultUnitLeft);
    expect(wrapper.find('Dropdown.unitRight').props().value).toBe(defaultUnitRight);
  });
});
