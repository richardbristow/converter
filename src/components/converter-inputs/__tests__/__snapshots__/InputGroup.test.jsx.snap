// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup should render an input element, with the correct props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputGroup
    conversionType="type1"
    dropdownValue="mathName2"
    handleChange={[Function]}
    name="name"
    options={
        Array [
            Object {
              "displayName": "display1",
              "mathName": "mathName1",
            },
            Object {
              "displayName": "display2",
              "mathName": "mathName2",
            },
            Object {
              "displayName": "display3",
              "mathName": "mathName3",
            },
          ]
    }
    textValue="InputValue"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="nameInput"
          onChange={[Function]}
          type="text"
          value="InputValue"
/>,
        <select
          name="nameUnit"
          onChange={[Function]}
          value="mathName2"
>
          <option
                    value="mathName1"
          >
                    display1
          </option>
          <option
                    value="mathName2"
          >
                    display2
          </option>
          <option
                    value="mathName3"
          >
                    display3
          </option>
</select>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "nameInput",
          "onChange": [Function],
          "type": "text",
          "value": "InputValue",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              value="mathName1"
>
              display1
</option>,
            <option
              value="mathName2"
>
              display2
</option>,
            <option
              value="mathName3"
>
              display3
</option>,
          ],
          "name": "nameUnit",
          "onChange": [Function],
          "value": "mathName2",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "name-mathName1",
            "nodeType": "host",
            "props": Object {
              "children": "display1",
              "value": "mathName1",
            },
            "ref": null,
            "rendered": "display1",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "name-mathName2",
            "nodeType": "host",
            "props": Object {
              "children": "display2",
              "value": "mathName2",
            },
            "ref": null,
            "rendered": "display2",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "name-mathName3",
            "nodeType": "host",
            "props": Object {
              "children": "display3",
              "value": "mathName3",
            },
            "ref": null,
            "rendered": "display3",
            "type": "option",
          },
        ],
        "type": "select",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="nameInput"
            onChange={[Function]}
            type="text"
            value="InputValue"
/>,
          <select
            name="nameUnit"
            onChange={[Function]}
            value="mathName2"
>
            <option
                        value="mathName1"
            >
                        display1
            </option>
            <option
                        value="mathName2"
            >
                        display2
            </option>
            <option
                        value="mathName3"
            >
                        display3
            </option>
</select>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "nameInput",
            "onChange": [Function],
            "type": "text",
            "value": "InputValue",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="mathName1"
>
                display1
</option>,
              <option
                value="mathName2"
>
                display2
</option>,
              <option
                value="mathName3"
>
                display3
</option>,
            ],
            "name": "nameUnit",
            "onChange": [Function],
            "value": "mathName2",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "name-mathName1",
              "nodeType": "host",
              "props": Object {
                "children": "display1",
                "value": "mathName1",
              },
              "ref": null,
              "rendered": "display1",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "name-mathName2",
              "nodeType": "host",
              "props": Object {
                "children": "display2",
                "value": "mathName2",
              },
              "ref": null,
              "rendered": "display2",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": "name-mathName3",
              "nodeType": "host",
              "props": Object {
                "children": "display3",
                "value": "mathName3",
              },
              "ref": null,
              "rendered": "display3",
              "type": "option",
            },
          ],
          "type": "select",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
